WooCommerce/Classes/ViewRelated/Products/Variations/ProductVariationsViewController.swift

```diff
@@ -14,15 +14,18 @@ final class ProductVariationsViewController: UIViewController {
 
     /// Empty state screen configuration
     ///
-    private lazy var emptyStateConfig: EmptyStateViewController.Config = {
+    private var emptyStateConfig: EmptyStateViewController.Config {
         let message = NSAttributedString(string: Localization.emptyStateTitle, attributes: [.font: EmptyStateViewController.Config.messageFont])
+        let shouldShowCreatingAttributesGuide = viewModel.shouldShowCreatingAttributesGuide(for: product)
+        let subtitle = shouldShowCreatingAttributesGuide ? Localization.emptyStateSubtitle : ""
+        let buttonTitle = shouldShowCreatingAttributesGuide ? Localization.addAttributesAction : Localization.addVariationAction
         return .withButton(message: message,
                            image: .emptyBoxImage,
-                           details: "",
-                           buttonTitle: Localization.emptyStateButtonTitle) { [weak self] _ in
+                           details: subtitle,
+                           buttonTitle: buttonTitle) { [weak self] _ in
                             self?.createVariationFromEmptyState()
                            }
-    }()
+    }
```

Comment:
Since this configuration now depends on a view model value, I think that having a function that creates it will express the intention better, as you can make the dependencies explicit.

```swift
func createEmptyStateConfig(showAttributeGuide: Bool) -> EmptyStateViewController.Config {
  /// code
}
```

Reply:
I think this is a nice suggestion! Will surely update it.